contracts/finance/VestingWalletConfidentialFactory.sol

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {FHE, euint64, externalEuint64, euint128, ebool} from "@fhevm/solidity/lib/FHE.sol";
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {IConfidentialFungibleToken} from "../interfaces/IConfidentialFungibleToken.sol";
+import {VestingWalletCliffConfidential} from "./VestingWalletCliffConfidential.sol";
+import {VestingWalletConfidential} from "./VestingWalletConfidential.sol";
+import {VestingWalletExecutorConfidential} from "./VestingWalletExecutorConfidential.sol";
+
+/**
+ * @dev This factory enables creating {VestingWalletCliffExecutorConfidential} in batch.
+ *
+ * All confidential vesting wallets created support both "cliff" ({VestingWalletCliffConfidential})
+ * and "executor" ({VestingWalletExecutorConfidential}) extensions.
+ */
+abstract contract VestingWalletConfidentialFactory {
```

Comment:
Does this need to be abstract? We intend for this file to be deployed as is right?

Reply:
It could be deployer as-is. See [Set factory as non abstract](https://github.com/OpenZeppelin/openzeppelin-confidential-contracts/pull/102/commits/ee6a464082714a83eb42fcc8c474e82be18b7a44).